src/generic/AbstractBandedMatrix.jl

```diff
@@ -401,3 +401,46 @@ function sum(A::AbstractBandedMatrix; dims=:)
         throw(ArgumentError("dimension must be ≥ 1, got $dims"))
     end
 end
+
+###
+# vcat
+###
+
+function LinearAlgebra.vcat(x::AbstractBandedMatrix...)
+    #avoid unnecessary steps for singleton
+    if length(x) == 1
+        return x[1]
+    end
+
+    #instantiate the returned banded matrix with zeros and required bandwidths/dimensions
+    m = size(x[1], 2)
+    n = 0
+
+    for A in x
+        if size(A, 2) != m
+            sizes = Tuple(size(b, 2) for b in x)
+            throw(DimensionMismatch("number of columns of each matrix must match (got $sizes)"))
+        end
+
+        n += size(A, 1)
+    end
+
+    l = n - size(x[end], 1) + bandwidth(x[end], 1)
+    u = bandwidth(x[1], 2)
+    type = promote_type(eltype.(x)...)
+    ret = BandedMatrix{type}(Zeros(n, m), (l, u))
```

Comment:
Since `type` might not be a number we shouldn't first make a `Zeros{Float64}` (which is the default). This is better:
```suggestion
    ret = BandedMatrix(Zeros{type}(n, m), (l, u))
```